api.lua

```diff
@@ -94,9 +94,13 @@ function areas:canInteract(pos, name)
 	for _, area in pairs(self:getAreasAtPos(pos)) do
 		if area.owner == name or area.open then
 			return true
-		else
-			owned = true
+		elseif factions and area.faction_open then
```

Comment:
same. consider caching the return value of `global_exists` outside this function